### PR TITLE
[d3-array] Update API definition to 3.2.4. Closes #65707

### LINKED ISSUE #65707: Declaration of medianIndex is missing in [@types/d3-array]
### Discussed in https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/65706

<div type='discussions-op-text'>

<sup>Originally posted by **smpa01** June  7, 2023</sup>
It seems that the d3.medianIndex is missing in @types/d3-array.
Please refer below
- https://github.com/d3/d3-array/blob/main/src/median.js
- https://github.com/d3/d3-array/blob/main/src/index.js#L21

</div>

### PR DESCRIPTION
**Pull request by danmana**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/d3/d3-array/releases/tag/v3.2.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Closes #65707

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## d3-array ([<kbd>unpkg</kbd>](https://unpkg.com/browse/d3-array@latest/))
was missing the following properties:
1. scan

## Formatting

The following files are not formatted:
1. types/d3-array/index.d.ts
1. types/d3-array/d3-array-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0bcfa5618837b7669466d896e7360603c7cb1806
</p>

**Comment by Fil** [comment]
@DangerBotOSS  d3.scan does not need to be typed because it's a (deprecated) alias of d3.leastIndex
ref. https://observablehq.com/@d3/d3v6-migration-guide

I don't know about formatting requirements.

File: types/d3-array/index.d.ts [hunk 1]
```diff
@@ -231,6 +253,19 @@ export function quantile<T>(
     accessor: (element: T, i: number, array: Iterable<T>) => number | undefined | null,
 ): number | undefined;
 
+/**
+ * Similar to quantile, but returns the index to the left of p.
+ */
+export function quantileIndex(iterable: Iterable<Numeric | undefined | null>, p: number): number | undefined;
```
Comment (Fil) [inline types/d3-array/index.d.ts:259]:
```suggestion
export function quantileIndex(iterable: Iterable<Numeric | undefined | null>, p: number): number;
```

File: types/d3-array/index.d.ts [hunk 2]
```diff
@@ -196,6 +206,18 @@ export function median<T>(
     accessor: (element: T, i: number, array: Iterable<T>) => number | undefined | null,
 ): number | undefined;
 
+/**
+ * Like median, but returns the index of the element to the left of the median.
+ */
+export function medianIndex(iterable: Iterable<Numeric | undefined | null>): number | undefined;
```
Comment (Fil) [inline types/d3-array/index.d.ts:212]:
```suggestion
export function medianIndex(iterable: Iterable<Numeric | undefined | null>): number;
```

these functions return -1 when there is no defined value in the iterable.

File: types/d3-array/index.d.ts [hunk 3]
```diff
@@ -196,6 +206,18 @@ export function median<T>(
...
+/**
+ * Like median, but returns the index of the element to the left of the median.
+ */
+export function medianIndex<T>(
+    iterable: Iterable<T>,
+    accessor: (element: T, i: number, array: Iterable<T>) => number | undefined | null,
+): number | undefined;
```
Comment (Fil) [inline types/d3-array/index.d.ts:219]:
```suggestion
): number;
```

File: types/d3-array/index.d.ts [hunk 4]
```diff
@@ -231,6 +253,19 @@ export function quantile<T>(
...
+ */
+export function quantileIndex(iterable: Iterable<Numeric | undefined | null>, p: number): number | undefined;
+/**
+ * Similar to quantile, but returns the index to the left of p.
+ */
+export function quantileIndex<T>(
+    iterable: Iterable<T>,
+    p: number,
+    accessor: (element: T, i: number, array: Iterable<T>) => number | undefined | null,
+): number | undefined;
```
Comment (Fil) [inline types/d3-array/index.d.ts:267]:
```suggestion
): number;
```

**Review by Fil — CHANGES_REQUESTED** [review]

**Review by Fil — APPROVED** [review]

**Comment by danmana** [comment]
Ready to merge

**Pull request merged**